File: courses/cs50x/psets/6/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    draw_pyramids(get_height())


def get_height() -> int:
    while True:
        h: int | None = get_int("Height: ")
        if h is not None and 1 <= h <= 8:
            return h


def draw_pyramids(h: int) -> None:
    spaces: int = h - 1
    for _ in range(h):
        print(f"{' ' * spaces}{'#' * (h - spaces)}  {'#' * (h - spaces)}")
        spaces -= 1


main()

File: courses/cs50x/psets/1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_row(int height, int spaces);

int main(void)
{
	int height = get_height();
	for (int i = 0; i < height; i++)
		print_row(height, height - i - 1);
	return 0;
}

int get_height(void)
{
	const int MIN_HEIGHT = 1;
	int h = 0;

	while (h < MIN_HEIGHT)
		h = get_int("Height: ");
	return h;
}

void print_row(int h, int s)
{
	for (int i = 0; i < h; i++)
	{
		if (i >= s)
			printf("#");
		else
			printf(" ");
	}
	printf("\n");
}
